pkg/types/container/v0.0.1/entry_test.go

```diff
@@ -0,0 +1,269 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package container
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"io/ioutil"
+	"reflect"
+	"testing"
+
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/sigstore/rekor/pkg/types"
+	"github.com/sigstore/sigstore/pkg/signature/payload"
+	"go.uber.org/goleak"
+)
+
+func TestMain(m *testing.M) {
+	goleak.VerifyTestMain(m)
+}
+
+func TestNewEntryReturnType(t *testing.T) {
+	entry := NewEntry()
+	if reflect.TypeOf(entry) != reflect.ValueOf(&V001Entry{}).Type() {
+		t.Errorf("invalid type returned from NewEntry: %T", entry)
+	}
+}
+
+func TestCrossFieldValidation(t *testing.T) {
+	type TestCase struct {
+		caseDesc                  string
+		entry                     V001Entry
+		expectUnmarshalSuccess    bool
+		expectCanonicalizeSuccess bool
+	}
+
+	sigBytes, _ := ioutil.ReadFile("../../../../tests/test_container.sig")
+	keyBytes, _ := ioutil.ReadFile("../../../../tests/test_container_public_key.key")
+	dataBytes, _ := ioutil.ReadFile("../../../../tests/test_container.txt")
+
+	h := sha256.Sum256(bytes.TrimSpace(dataBytes))
+	dataSHA := hex.EncodeToString(h[:])
+
+	dataContent := &payload.Cosign{}
+	_ = json.Unmarshal(dataBytes, dataContent)
+
+	testCases := []TestCase{
```

Comment:
fyi: 

```
bcallaway@bcallaway01:~/devconf$ od -c junk2
0000000   n   u   l   l
0000004
bcallaway@bcallaway01:~/devconf$ /tmp/rekor/rekor-cli upload --artifact junk2 --signature junk2.sig --public-key ec_public.pem --type container --rekor_server=http://localhost:3000
Created entry at index 0, available at: http://localhost:3000/api/v1/log/entries/6e3840b58e13be114a8c83e1dace1731503ffaff357a079b7f70d28186933bc5
```

due to https://github.com/sigstore/sigstore/blob/10daa6daa06d88ce963939db227228764a336595/pkg/signature/payload/payload.go#L83